### PR TITLE
Transcode features

### PR DESCRIPTION
**Pull request by bmatherly**
Use ffmpeg's minterpolate filter to provide:
* duplicate
* blend
* Motion estimation

image

### CODE REVIEW

File: src/widgets/avformatproducerwidget.cpp [hunk 1]
```diff
@@ -782,10 +782,28 @@ void AvformatProducerWidget::convert(TranscodeDialog& dialog)
...
+
+        // Set video filters
+        args << "-vf";
+        QString filterString;
+        QString range;
+        if (ui->rangeComboBox->currentIndex())
+            range = "full";
+        else
+            range = "mpeg";
+        filterString = QString("scale=flags=accurate_rnd+full_chroma_inp+full_chroma_int:in_range=full:out_range=%1").arg(range);
```
Comment (ddennedy) [inline src/widgets/avformatproducerwidget.cpp:794]: Why does this change assume the in_range is full? It should use the same as out_range.

File: src/dialogs/transcodedialog.ui [hunk 2]
```diff
@@ -140,6 +137,61 @@
...
+         <widget class="QLabel" name="fpsLabel">
+          <property name="text">
+           <string>Frames/sec</string>
+          </property>
+         </widget>
+        </item>
+        <item row="3" column="0">
+         <widget class="QLabel" name="frcLabel">
+          <property name="text">
+           <string>Frame Rate Conversion</string>
```
Comment (ddennedy) [inline src/dialogs/transcodedialog.ui:179]: Please change the casing on this label to uppercase first word only. See https://shotcut.org/notes/ui-conventions/

File: src/dialogs/transcodedialog.ui [hunk 3]
```diff
@@ -140,6 +137,61 @@
...
+     <layout class="QVBoxLayout" name="verticalLayout_2">
+      <item>
+       <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
+        <item row="1" column="1">
+         <widget class="QCheckBox" name="fpsCheckBox">
+          <property name="toolTip">
+           <string>Enable frame rate override.</string>
+          </property>
+          <property name="text">
+           <string>Override Frame Rate</string>
```
Comment (ddennedy) [inline src/dialogs/transcodedialog.ui:151]: Please change the casing on this label to uppercase first word only. See https://shotcut.org/notes/ui-conventions/

**Review by ddennedy — CHANGES_REQUESTED** [review]

File: src/widgets/avformatproducerwidget.cpp (hunk 1 above)
Comment (bmatherly) [reply src/widgets/avformatproducerwidget.cpp]: Typo. Fixed.

File: src/dialogs/transcodedialog.ui (hunk 2 above)
Comment (bmatherly) [reply src/dialogs/transcodedialog.ui]: fixed

File: src/dialogs/transcodedialog.ui (hunk 3 above)
Comment (bmatherly) [reply src/dialogs/transcodedialog.ui]: fixed

File: scripts/build-shotcut.sh [hunk 4]
```diff
@@ -410,6 +417,9 @@ function set_globals {
     if test "$ENABLE_BIGSH0T" = 1 ; then
         SUBDIRS="$SUBDIRS bigsh0t"
     fi
+    if test "$ENABLE_ZIMG" = 1 ; then
+        SUBDIRS="$SUBDIRS zimg"
```
Comment (ddennedy) [inline scripts/build-shotcut.sh:421]: This needs to be `SUBDIRS="zimg $SUBDIRS"` so that FFmpeg build can see it.

Comment (ddennedy) [reply scripts/build-shotcut.sh:421]: I can fix this order un SUBDIRS in merge.  I made changes for the windows build and testing it now.

**Pull request merged**